src/app/navigation/hooks/useBreadcrumbs.ts

```diff
@@ -0,0 +1,34 @@
+/***********************************************************
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License
+ **********************************************************/
+import * as React from 'react';
+import { BreadcrumbEntry } from '../model';
+
+export interface UseBreadcrumbsResult {
+    stack: BreadcrumbEntry[];
+    registerEntry(entry: BreadcrumbEntry): void;
+    unregisterEntry(entry: BreadcrumbEntry): void;
+}
+
+export const useBreadcrumbs = (): UseBreadcrumbsResult => {
+    const [stack, setStack] = React.useState<BreadcrumbEntry[]>([]);
+
+    const registerEntry = React.useCallback((entry: BreadcrumbEntry) => {
+        setStack(currentState => {
+            const newStack = [...currentState.filter(s => entry.path !== s.path), entry];
+            const sortedStack = newStack.sort((a, b) => a.path.length - b.path.length);
```

Comment:
not quite sure if i understand the reason behind this sorting?

Reply:
Sure thing -- the sort is necessary to handle component mount / unmount.  Component mount order is from the bottom up -- so without the sorting -- a load that adds multiple nested components would actually be out of order.  The sort ensures longer paths get put lower in the breadcrumb.